src/game/WorldSocket.cpp

```diff
@@ -258,496 +131,141 @@ int WorldSocket::open(void* a)
-            {
-                // Couldn't receive the whole data this time.
-                MANGOS_ASSERT(message_block.length() == 0);
-                errno = EWOULDBLOCK;
-                return -1;
-            }
-        }
-
-        // just received fresh new payload
-        if (handle_input_payload() == -1)
-        {
-            MANGOS_ASSERT((errno != EWOULDBLOCK) && (errno != EAGAIN));
-            return -1;
-        }
-    }
-
-    return size_t(n) == recv_size ? 1 : 2;
-}
-
-int WorldSocket::cancel_wakeup_output(GuardType& g)
-{
-    if (!m_OutActive)
-        return 0;
-
-    m_OutActive = false;
-
-    // FIXME: I think it is okay to let this be released when it goes out of scope?
-    // If we call the destructor here, the behavior when it is called again upon going
-    // out of scope is undefined.
-    //g.release();
-
-    if (reactor()->cancel_wakeup
-            (this, ACE_Event_Handler::WRITE_MASK) == -1)
-    {
-        // would be good to store errno from reactor with errno guard
-        sLog.outError("WorldSocket::cancel_wakeup_output");
-        return -1;
+        errno = EBADMSG;
+        return false;
     }
 
-    return 0;
-}
+    const Opcodes opcode = static_cast<Opcodes>(header.cmd);
 
-int WorldSocket::schedule_wakeup_output(GuardType& g)
-{
-    if (m_OutActive)
-        return 0;
+    if (IsClosed())
+        return false;
 
-    m_OutActive = true;
-
-    // FIXME: I think it is okay to let this be released when it goes out of scope?
-    // If we call the destructor here, the behavior when it is called again upon going
-    // out of scope is undefined.
-    //g.release();
-
-    if (reactor()->schedule_wakeup
-            (this, ACE_Event_Handler::WRITE_MASK) == -1)
-    {
-        sLog.outError("WorldSocket::schedule_wakeup_output");
-        return -1;
-    }
-
-    return 0;
-}
-
-int WorldSocket::ProcessIncoming(WorldPacket* new_pct)
-{
-    MANGOS_ASSERT(new_pct);
-
-    // manage memory ;)
-    ACE_Auto_Ptr<WorldPacket> aptr(new_pct);
+    // Dump received packet.
+    //sLog.outWorldPacketDump(uint32(get_handle()), new_pct->GetOpcode(), new_pct->GetOpcodeName(), new_pct, true);
 
-    const ACE_UINT16 opcode = new_pct->GetOpcode();
+    WorldPacket *pct = new WorldPacket(opcode, validBytesRemaining);
 
-    if (opcode >= NUM_MSG_TYPES)
+    if (validBytesRemaining)
     {
-        sLog.outError("SESSION: received nonexistent opcode 0x%.4X", opcode);
-        return -1;
+        pct->append(InPeak(), validBytesRemaining);
+        ReadSkip(validBytesRemaining);
     }
 
-    if (closing_)
-        return -1;
-
-    // Dump received packet.
-    sLog.outWorldPacketDump(uint32(get_handle()), new_pct->GetOpcode(), new_pct->GetOpcodeName(), new_pct, true);
-
     try
     {
         switch (opcode)
         {
-            case CMSG_PING:
-                return HandlePing(*new_pct);
             case CMSG_AUTH_SESSION:
-                if (m_Session)
+                if (!!m_session)
```

Comment:
It was to silence a compiler warning from Visual Studio and clang which complain about a performance hit in casting an integer type to a boolean.  I mistakenly thought that that applied to pointers, as well.  It will be removed from the final merge.